File: vm/engine/java/src/test/java/dev/argon/argonvm/engine/TestStandardLibrary.java
package dev.argon.argonvm.engine;

import dev.argon.argonvm.StandardLibrary;

public class TestStandardLibrary extends StandardLibrary {

	public final StringBuilder output = new StringBuilder();

	@Override
	public void print(String value) throws Throwable {
		output.append(value);
	}
}

File: argon-vm-engine/js-node/src/test/scala/dev/argon/vm/engine/TestStandardLibrary.scala
package dev.argon.vm.engine

import zio.*
import typings.argonLangVmEngine.mod.StandardLibrary

import scala.collection.mutable

class TestStandardLibrary(output: mutable.StringBuilder) extends StandardLibrary {
  def getOutput: UIO[String] = ZIO.succeed { output.toString }

  override def print(value: String): Unit = {
    output.append(value)
  }

  def toNativeFunctions: NativeFunctions = nativeFunctions

}

object TestStandardLibrary {
  def make: UIO[TestStandardLibrary] =
    for
      output <- ZIO.succeed { new StringBuilder() }
    yield TestStandardLibrary(output)
}

